File: src/main/java/gargoyle/rpycg/fx/FXSplash.java
package gargoyle.rpycg.fx;

import java.awt.Window;
import java.net.URL;
import java.util.ResourceBundle;

@FunctionalInterface
public interface FXSplash {

    Window createWindow(URL location, ResourceBundle resources);

    default void handleSplashNotification(final FXSplashNotification splashNotification) {
    }

    interface FXSplashNotification {
        String getDetails();

        double getProgress();

        Type getType();

        enum Type {
            PRE_INIT,
            INIT,
            START,
            STOP
        }
    }
}

File: fx/src/main/kotlin/gargoyle/fx/FxSplash.kt
package gargoyle.fx

import java.awt.Window
import java.net.URL
import java.util.ResourceBundle

fun interface FxSplash {
    fun createWindow(location: URL, resources: ResourceBundle?): Window
    fun handleSplashNotification(splashNotification: FxSplashNotification) {}
    interface FxSplashNotification {
        val details: String
        val progress: Double
        val type: Type

        enum class Type {
            PRE_INIT, INIT, START, STOP
        }
    }
}
